pkg/lifecycle/startup/startup.go

```diff
@@ -772,6 +809,27 @@ func autoDetectCIDRBySkipInterface(ifaceRegexes []string, version int) *cnet.IPN
 	return cidr
 }
 
+// autoDetectUsingK8sInternalIP reads K8s Node InternalIP.
+func autoDetectUsingK8sInternalIP(version int, k8sNode *v1.Node) *cnet.IPNet {
+	var address string
+	var err error
+
+	nodeAddresses := k8sNode.Status.Addresses
+	for _, addr := range nodeAddresses {
+		if addr.Type == v1.NodeInternalIP {
+			if (version == 4 && utils.IsIPv4String(addr.Address)) || (version == 6 && utils.IsIPv6String(addr.Address)) {
+				address, err = getLocalCIDR(addr.Address, version, autodetection.GetInterfaces)
+			}
+			if err != nil {
+				return nil
+			}
+		}
+	}
+
+	ipNet := cnet.MustParseCIDR(address)
```

Comment:
We should use `cnet.ParseCIDR` instead of `MustParseCIDR` - the latter is just for testing, and can result in panics. We want to handle any error gracefully.

Reply:
modified as per suggestion